cmd/block-generator/generator/generate.go

```diff
@@ -290,35 +371,193 @@ func (g *generator) generatePaymentTxn(round uint64, intra uint64) (types.Signed
+			util.MaybeFail(err, "unable to make asset destroy transaction")
+
+			asset.holdings = append(asset.holdings, assetHolding{
+				acctIndex: senderIndex,
+				balance:   0,
+			})
+			asset.holders[senderIndex] = true
+		case assetXfer:
+			// send from creator (holder[0]) to another random holder (same address is valid)
+
+			// If there aren't enough assets to close one, optin an account instead
+			if len(asset.holdings) == 1 {
+				return g.generateAssetTxnInternalHint(assetOptin, sp, intra, assetIndex, asset)
+			}
+
+			senderIndex = asset.holdings[0].acctIndex
+			senderString := indexToAccount(senderIndex).String()
+
+			receiverIndex := (rand.Uint64() % (uint64(len(asset.holdings)) - uint64(1))) + uint64(1)
+			receiver := indexToAccount(asset.holdings[receiverIndex].acctIndex)
+			receiverString := receiver.String()
+
+			txn, err = future.MakeAssetTransferTxn(senderString, receiverString, 10, nil, sp, "", asset.assetID)
+			util.MaybeFail(err, "unable to make asset destroy transaction")
+
+			if asset.holdings[0].balance < 10 {
+				fmt.Printf(fmt.Sprintf("\n\ncreator doesn't have enough funds for asset %d\n\n", asset.assetID))
+				os.Exit(1)
+			}
+
+			asset.holdings[0].balance -= 10
+			asset.holdings[receiverIndex].balance += 10
+		case assetClose:
+			// select a holder of a random asset to close out
+			// If there aren't enough assets to close one, optin an account instead
+			if len(asset.holdings) == 1 {
+				return g.generateAssetTxnInternalHint(assetOptin, sp, intra, assetIndex, asset)
+			}
+
+			numHoldings := uint64(len(asset.holdings))
+			closeIndex := (rand.Uint64() % (numHoldings - 1)) + uint64(1)
+			senderIndex = asset.holdings[closeIndex].acctIndex
+			senderString := indexToAccount(senderIndex).String()
+
+			closeToAcctIndex := asset.holdings[0].acctIndex
+			closeToAcct := indexToAccount(closeToAcctIndex)
+			closeToAcctString := closeToAcct.String()
+
+			txn, err = future.MakeAssetTransferTxn(senderString, closeToAcctString, 0, nil, sp, closeToAcctString, asset.assetID)
+			util.MaybeFail(err, "unable to make asset destroy transaction")
+
+			asset.holdings[0].balance += asset.holdings[closeIndex].balance
+
+			// Remove asset by swapping the element to delete then trimming the last
+			asset.holdings[numHoldings-1], asset.holdings[closeIndex] = asset.holdings[closeIndex], asset.holdings[numHoldings-1]
+			asset.holdings = asset.holdings[:numHoldings-1]
+			delete(asset.holders, senderIndex)
+		default:
+		}
+	}
+
+	if indexToAccount(senderIndex) != txn.Sender {
+		fmt.Printf("failed to properly set sender index.")
+		os.Exit(1)
+	}
+
+	g.balances[senderIndex] -= uint64(txn.Fee)
+	return
+}
+
+func (g *generator) generateAssetTxn(sp sdk_types.SuggestedParams, _ uint64, intra uint64) (types.SignedTxnInBlock, error) {
+	if len(g.assetTxWeights) == 0 {
+		g.assetTxWeights = append(g.assetTxWeights, g.config.AssetCreateFraction)
```

Comment:
I'll move this to the Make function. I wanted to keep the initializing of this close to the `get___Options` functions, but I can initialize them deterministically to avoid needing to ensure the order is correct.